pangeo_forge_recipes/recipes/base.py

```diff
@@ -95,6 +93,26 @@ def __post_init__(self):
         # aren't relayed to `object`
         pass
 
+    def to_dask(self):
+        cls_ = self.__class__
+
+        def self_task():
+            return self
+
+        self_ = delayed(self_task)()
+
+        layers = [
+            [delayed(cls_.cache_input)(self_, input_key) for input_key in self.iter_inputs()],
```

Comment:
Dask experts @TomAugspurger / @martindurant: does this look right / reasonable? Is there a way to make this a HighLevelGraph layer?

I tried using dask bag, e.g. 
```python
db.from_sequence(range(self.iter_inputs()).map(cls_.cache_input, self_)
```

but I couldn't get around the fact that the _non-mapped_ argument (`self_` has to come after the mapped one in the function signature.

Reply:
This feels overly complicated. Give me a bit to play with it.